src/main/java/io/neonbee/internal/verticle/ServerVerticle.java

```diff
@@ -98,66 +123,56 @@ public void start(Promise<Void> startPromise) {
-                        Optional.ofNullable(NeonBee.get(vertx).getOptions().getServerPort()).ifPresent(config::setPort);
-
-                        vertx.createHttpServer(config /* ServerConfig is a HttpServerOptions subclass */)
-                                .exceptionHandler(throwable -> {
-                                    LOGGER.error("HTTP Socket Exception", throwable);
-                                }).requestHandler(router).listen().onSuccess(httpServer -> {
-                                    this.httpServer = httpServer;
-                                    if (LOGGER.isInfoEnabled()) {
-                                        LOGGER.info("HTTP server started on port {}", httpServer.actualPort());
-                                    }
-                                    if (LOGGER.isDebugEnabled()) {
-                                        LOGGER.debug("HTTP server configured with routes: {}", router.getRoutes()
-                                                .stream().map(Route::toString).collect(Collectors.joining(",")));
-                                    }
-                                }).onFailure(cause -> {
-                                    LOGGER.error("HTTP server could not be started", cause);
-                                }).<Void>mapEmpty().onComplete(startPromise);
-                    });
-        } catch (Exception e) {
-            LOGGER.error("Server could not be started", e);
-            startPromise.fail(e);
-        }
-    }
-
-    @Override
-    public void stop(Promise<Void> stopPromise) throws Exception {
-        // Vert.x would close the HTTP server for us, however we would like to do some additional logging
-        (httpServer != null ? httpServer.close().onComplete(result -> {
-            LOGGER.info("HTTP server was stopped");
-        }) : succeededFuture().<Void>mapEmpty()).onComplete(stopPromise);
+            return succeededFuture(router);
+        }).onFailure(e -> {
+            LOGGER.error("Router could not be created", e);
+        });
     }
 
     @VisibleForTesting
     // reason for PMD.EmptyCatchBlock empty cache block is "path-through" to fallback implementation and for
     // PMD.SignatureDeclareThrowsException it does not matter, if this private method does not expose any concrete
     // exceptions as the ServerVerticle start method anyways catches all exceptions in a generic try-catch block
     @SuppressWarnings({ "PMD.EmptyCatchBlock", "PMD.SignatureDeclareThrowsException" })
-    static ErrorHandler createErrorHandler(String className, String template) throws Exception {
-        Class<?> classObject = Class.forName(Optional.ofNullable(className).filter(Predicate.not(String::isBlank))
-                .orElse(DEFAULT_ERROR_HANDLER_CLASS_NAME));
-        if (template != null) {
+    static Future<ErrorHandler> createErrorHandler(String className, Vertx vertx) {
+        return AsyncHelper.executeBlocking(vertx, p -> {
```

Comment:
Is a executeBlcoking still required here?

Calling the constructor should be pretty quick and the eh.initialize method is now asynchronous, thus I would think we can remove the blocking call to not cause worker threads to be utilized even more during boot.